### ISSUE TITLE
No example from https://bastion.rs working

### ISSUE THREAD

**Issue by senden9** [bastion-rs/bastion#83]
Hi!
None of the examples from https://bastion.rs/#one is working for me. I create a common issue here because I think it is no coincidence that _both_ are broken. One does not compile, the other give a run-time error.

## Common
Cargo.toml:
```toml
[package]
name = "hello-world"
version = "0.1.0"
authors = ["Stefano Probst <senden9@gmail.com>"]
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
fort = "0.2.0"
bastion = "0.2.0"
``` 

Software versions:
```
$ cargo version
cargo 1.39.0 (1c6ec66d5 2019-09-30)

$ echo $PWD
/tmp/hello-world

$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 19.10
Release:	19.10
Codename:	eoan

$ uname -srv
Linux 5.3.0-19-generic #20-Ubuntu SMP Fri Oct 18 09:04:39 UTC 2019
```

## Example "configure every piece by yourself"
Code:
```rust
use bastion::prelude::*;

fn main() {
    Bastion::platform();

    // Define, calculate or prepare messages to be sent to the processes. 
    let message = String::from("Some message to be passed");

    Bastion::spawn(
        |context: BastionContext, msg: Box<dyn Message>| {
            // Message can be selected with receiver here. Take action!
            receive! { msg,
                String => |e| { println!("Received string :: {}", e)},
                i32 => |e| {println!("Received i32 :: {}", e)},
                _ => println!("No message as expected. Default")
            }

            // Do some processing in body
            println!("root supervisor - spawn_at_root - 1");

            // Rebind to the system
            context.hook();
        },
        message,
    );

    Bastion::start()
}
```
Error messages:
```
$ RUST_BACKTRACE=1 cargo run
   Compiling hello-world v0.1.0 (/tmp/hello-world)
error[E0433]: failed to resolve: use of undeclared type or module `objekt`
  --> src/main.rs:12:13
   |
12 | /             receive! { msg,
13 | |                 String => |e| { println!("Received string :: {}", e)},
14 | |                 i32 => |e| {println!("Received i32 :: {}", e)},
15 | |                 _ => println!("No message as expected. Default")
16 | |             }
   | |_____________^ use of undeclared type or module `objekt`
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0433`.
error: could not compile `hello-world`.

To learn more, run the command again with --verbose.
```

## Example "fort the proc-macro"
```rust
#[fort::root]
fn main() {
    println!("Running in Bastion runtime!");
}
```

Run-time Error message:
<details>
  <summary>Long backtrace inside</summary>

```
$ RUST_BACKTRACE=1 cargo run 2> error.log
[break code via ctrl+c]
$ head -n 180 error.log
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/hello-world`
thread 'tokio-runtime-worker-0' panicked at 'called `Option::unwrap()` on a `None` value', src/libcore/option.rs:378:21
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.37/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.37/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:76
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:60
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1030
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1412
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:64
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:196
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:210
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:473
  11: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:380
  12: rust_begin_unwind
             at src/libstd/panicking.rs:307
  13: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
  14: core::panicking::panic
             at src/libcore/panicking.rs:49
  15: core::option::Option<T>::unwrap
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libcore/macros.rs:12
  16: bastion::context::BastionContext::spawn
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/bastion-0.2.0/src/context.rs:213
  17: hello_world::main::{{closure}}
             at src/main.rs:1
  18: <alloc::boxed::Box<F> as core::ops::function::Fn<A>>::call
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/liballoc/boxed.rs:936
  19: bastion::bastion::Bastion::spawn::{{closure}}
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/bastion-0.2.0/src/bastion.rs:401
  20: futures::future::lazy::Lazy<F,R>::get
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.29/src/future/lazy.rs:64
  21: <futures::future::lazy::Lazy<F,R> as futures::future::Future>::poll
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.29/src/future/lazy.rs:82
  22: futures::future::catch_unwind::<impl futures::future::Future for std::panic::AssertUnwindSafe<F>>::poll
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.29/src/future/catch_unwind.rs:49
  23: <futures::future::catch_unwind::CatchUnwind<F> as futures::future::Future>::poll::{{closure}}
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.29/src/future/catch_unwind.rs:32
  24: std::panicking::try::do_call
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/panicking.rs:292
  25: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:80
  26: std::panicking::try
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/panicking.rs:271
  27: std::panic::catch_unwind
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/panic.rs:394
  28: <futures::future::catch_unwind::CatchUnwind<F> as futures::future::Future>::poll
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.29/src/future/catch_unwind.rs:32
  29: futures::future::chain::Chain<A,B,C>::poll
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.29/src/future/chain.rs:26
  30: <futures::future::then::Then<A,B,F> as futures::future::Future>::poll
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.29/src/future/then.rs:32
  31: <alloc::boxed::Box<F> as futures::future::Future>::poll
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.29/src/future/mod.rs:113
  32: futures::task_impl::Spawn<T>::poll_future_notify::{{closure}}
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.29/src/task_impl/mod.rs:329
  33: futures::task_impl::Spawn<T>::enter::{{closure}}
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.29/src/task_impl/mod.rs:399
  34: futures::task_impl::std::set
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.29/src/task_impl/std/mod.rs:83
  35: futures::task_impl::Spawn<T>::enter
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.29/src/task_impl/mod.rs:399
  36: futures::task_impl::Spawn<T>::poll_fn_notify
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.29/src/task_impl/mod.rs:291
  37: futures::task_impl::Spawn<T>::poll_future_notify
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.29/src/task_impl/mod.rs:329
  38: tokio_threadpool::task::Task::run::{{closure}}
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.16/src/task/mod.rs:145
  39: core::ops::function::FnOnce::call_once
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libcore/ops/function.rs:227
  40: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/panic.rs:315
  41: std::panicking::try::do_call
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/panicking.rs:292
  42: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:80
  43: std::panicking::try
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/panicking.rs:271
  44: std::panic::catch_unwind
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/panic.rs:394
  45: tokio_threadpool::task::Task::run
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.16/src/task/mod.rs:130
  46: tokio_threadpool::worker::Worker::run_task2
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.16/src/worker/mod.rs:567
  47: tokio_threadpool::worker::Worker::run_task
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.16/src/worker/mod.rs:459
  48: tokio_threadpool::worker::Worker::try_run_owned_task
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.16/src/worker/mod.rs:390
  49: tokio_threadpool::worker::Worker::try_run_task
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.16/src/worker/mod.rs:297
  50: tokio_threadpool::worker::Worker::run
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.16/src/worker/mod.rs:241
  51: tokio::runtime::threadpool::builder::Builder::build::{{closure}}::{{closure}}::{{closure}}::{{closure}}
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.22/src/runtime/threadpool/builder.rs:390
  52: tokio_timer::timer::handle::with_default::{{closure}}
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-timer-0.2.11/src/timer/handle.rs:101
  53: std::thread::local::LocalKey<T>::try_with
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/thread/local.rs:262
  54: std::thread::local::LocalKey<T>::with
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/thread/local.rs:239
  55: tokio_timer::timer::handle::with_default
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-timer-0.2.11/src/timer/handle.rs:84
  56: tokio::runtime::threadpool::builder::Builder::build::{{closure}}::{{closure}}::{{closure}}
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.22/src/runtime/threadpool/builder.rs:382
  57: tokio_timer::clock::clock::with_default::{{closure}}
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-timer-0.2.11/src/clock/clock.rs:137
  58: std::thread::local::LocalKey<T>::try_with
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/thread/local.rs:262
  59: std::thread::local::LocalKey<T>::with
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/thread/local.rs:239
  60: tokio_timer::clock::clock::with_default
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-timer-0.2.11/src/clock/clock.rs:117
  61: tokio::runtime::threadpool::builder::Builder::build::{{closure}}::{{closure}}
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.22/src/runtime/threadpool/builder.rs:381
  62: tokio_reactor::with_default::{{closure}}
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-reactor-0.1.10/src/lib.rs:237
  63: std::thread::local::LocalKey<T>::try_with
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/thread/local.rs:262
  64: std::thread::local::LocalKey<T>::with
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/thread/local.rs:239
  65: tokio_reactor::with_default
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-reactor-0.1.10/src/lib.rs:217
  66: tokio::runtime::threadpool::builder::Builder::build::{{closure}}
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.1.22/src/runtime/threadpool/builder.rs:380
  67: tokio_threadpool::callback::Callback::call
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.16/src/callback.rs:22
  68: tokio_threadpool::worker::Worker::do_run::{{closure}}::{{closure}}
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.16/src/worker/mod.rs:127
  69: tokio_executor::global::with_default::{{closure}}
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-executor-0.1.8/src/global.rs:209
  70: std::thread::local::LocalKey<T>::try_with
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/thread/local.rs:262
  71: std::thread::local::LocalKey<T>::with
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/thread/local.rs:239
  72: tokio_executor::global::with_default
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-executor-0.1.8/src/global.rs:178
  73: tokio_threadpool::worker::Worker::do_run::{{closure}}
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.16/src/worker/mod.rs:125
  74: std::thread::local::LocalKey<T>::try_with
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/thread/local.rs:262
  75: std::thread::local::LocalKey<T>::with
             at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/thread/local.rs:239
  76: tokio_threadpool::worker::Worker::do_run
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.16/src/worker/mod.rs:116
  77: tokio_threadpool::pool::Pool::spawn_thread::{{closure}}
             at /home/stefano/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.16/src/pool/mod.rs:345
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
[BASTION] - [ERROR] - Panic happened in root level child - Any
thread 'tokio-runtime-worker-0' panicked at 'called `Option::unwrap()` on a `None` value', src/libcore/option.rs:378:21
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.37/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.37/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:76
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:60
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1030
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1412
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:64
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
```

</details>

## Conclusion
I have no idea if this two things are connected as this was litaraly my first try of bastion/fort.
The demo project from above can be found under[hello-world.zip](https://github.com/bastion-rs/bastion/files/3828902/hello-world.zip). I deleted the `target` folder because otherwies this would be a few hundred megabyte.

**Comment by azastrael** [comment]
Hey @senden9 :wave:!

First of all, I'm very sorry for the inconvenience! We were able to find the origins of the two issues (which weren't connected) and they both have been fixed by `0.3.0` which will be released in the following days :wink:

`0.3.0` will have a lot of new, exiting features and I encourage you to look at our [two examples](https://github.com/bastion-rs/bastion/tree/master/bastion/examples) for it and at its documentation (`git clone https://github.com/bastion-rs/bastion && cd bastion/bastion && cargo doc --no-deps`) which is *almost* complete :smile:

Don't hesitate to create an other issue or contact us on discord if you find another issue or need help with anything :smiley:

**Comment by senden9** [comment]
Again on GitHub: Thanks, that was fast :D.

This examples are working. I will play around a bit with it. If I find something strange I will report!
